config/config.go

```diff
@@ -940,6 +956,40 @@ func ParseConfig(ctx *ParsingContext, file *hclparse.File, includeFromChild *Inc
 	return config, nil
 }
 
+// detectInputsCtyUsage detects if an identifier matching dependency.foo.inputs.bar is used in the given HCL file.
+//
+// This is deprecated functionality, so we look for this to determine if we should throw an error or warning.
+func detectInputsCtyUsage(file *hclparse.File) bool {
+	body, ok := file.Body.(*hclsyntax.Body)
+	if !ok {
+		return false
+	}
+
+	for _, attr := range body.Attributes {
+		for _, traversal := range attr.Expr.Variables() {
+			const dependencyInputsIdentifierMinParts = 3
+
+			if len(traversal) < dependencyInputsIdentifierMinParts {
+				continue
+			}
+
+			root, ok := traversal[0].(hcl.TraverseRoot)
+			if !ok || root.Name != "dependency" {
+				continue
+			}
+
+			attrTraversal, ok := traversal[2].(hcl.TraverseAttr)
```

Comment:
Do you need to check if the index exists?

Reply:
We already know that it's a slice larger than 3 above:
https://github.com/gruntwork-io/terragrunt/pull/3879/files/2d5e986605324b8e16a38a4294c1972ac047780f#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17R972